Tests/LibC/TestMkDir.cpp

```diff
@@ -103,3 +101,15 @@ TEST_CASE(unveil)
     res = unveil(nullptr, nullptr);
     EXPECT(res == 0);
 }
+
+TEST_CASE(pledge_without_cpath)
+{
+    int res = pledge("stdio proc", nullptr);
```

Comment:
We should do the limiting pledge here inside the child process. By doing it here in the parent process (EXPECT_CRASH forks a child), a future test randomization feature (i.e. running them in a randomized order) could break this test file.

Reply:
Interesting, gotcha

Are there any test examples where you fork a child process?

Reply:
The one you wrote already forks a child process

Reply:
This won't work because pledges/veils are only reset on `execve()`, not on `fork()`. `EXPECT_CRASH` forks a child, but doesn't `execve()`. We had this discussion already in #8643, #12462.

Disclaimer: It's been a long time since I worked on this, but I though I will pitch in because I have written that `FIXME` after running into the exact same problem.